### ISSUE TITLE
Effects must contain only functions

### ISSUE THREAD

**Issue by DarkLegend** [rematch/rematch#342]
Hey guys,

currently playing around with rematch to see if it would help us to give our project a better structure.
We use Typescript and everything is really object-oriented. That's one of the main reasons we look into rematch because Redux with all its functions feels a little wierd in that setup :-)

Some of our models are quite complex and we would love to see effects & reduceres separated.
Something like:

```javascript
@injectable() //We use inversify
export class UiModel implements Model {
    state: UiData = defaultUiState;
    reducers: any;
    effects: UiEffects;

    constructor(
        @inject(Symbols.fieldRegistry) fieldRegistry: FieldRegistry,
        @inject(Symbols.templateService) templateService: TemplateService
    ) {
        this.reducers = {}; //Does not matter for the example
        this.effects = new UiEffects(fieldRegistry, templateService);
    }
}

export class UiEffects {
    [id: string]: any;

    private fieldRegistry: FieldRegistry;
    private templateService: TemplateService;

    constructor(fieldRegistry: FieldRegistry, templateService: TemplateService) {
        this.fieldRegistry = fieldRegistry;
        this.templateService = templateService;
    }
   //Many Reducers
}
```

Downside: you currently MUST have only functions on the effects object. Our private vars are not allowed.
Any comments if my example is bad?
Or could we just remove the check and just ignore non-functions? Looks like it's any easy fix and I'm happy to provide a PR but wanted to hear your thoughts before..

Best Regards,
Andreas

**Comment by ShMcK** [comment]
Hi Andreas,

Glad to see you're trying to use Rematch with what looks like Angular 2. 

In this case, effects currently have to be an object with the format: `[key: string]: (payload, state):any`, I see you're trying to use `class`es instead. I think what you're trying to do may be doable without using classes.

Often when devs come to JS from other OOP languages, they bring a lot of preconceived notions. JS has classes, but they're not true classes. There's a great video series on objects, classes and prototypical inheritance at [Fun Fun Function](https://www.youtube.com/watch?v=GhbhD1HR5vk&list=PL0zVEGEvSaeHBZFy6Q8731rcwk0Gtuxub) which I personally learned a lot from, even after 5+ years of working with JS.

In short, I think what you're trying to do looks beautiful and well structured, though it could be greatly simplified if it dealt with simple objects rather than classes.

1) a simple object instead of a class. You can spread or merge items onto it.  An example:

```js
const model = {
  ...fieldRegistry,
  ...templateService,
}
```

2) if it is reused, a function (factory) that generates the object. An example:

```js
const uiEffectsGenerator = (fieldRegistry: FieldRegistry, templateService: TemplateService) => {
  return {
    // model here
  }
}
```

As a rule of thumb, I generally go by the following:

> If a class has a single new instance inside of it, that inner class should probably just be an object
